packages/nextjs/app/builders/page.tsx

```diff
@@ -0,0 +1,217 @@
+        console.log("Found builder profiles:", KNOWN_PROFILES);
+      } catch (error) {
+        console.error("Error checking builder profiles:", error);
+      }
+    };
+
+    checkBuilderProfiles();
+  }, []);
+
+  useEffect(() => {
+    if (elementRef.current) {
+      setIsLoading(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    const fetchBuilders = async () => {
+      if (!events) return;
+
+      // Filter out duplicate addresses
+      const filteredAddresses: string[] = [];
+      events.forEach(event => {
+        const address = event.args.builder;
+        if (!address) return;
+        if (!filteredAddresses.find(addr => address === addr)) filteredAddresses.push(address);
+      });
+
+      const builders = await Promise.all(
+        filteredAddresses.map(async address => {
+          if (address)
+            return {
+              address,
+              // Check if the builder has a personal page
+              hasPersonalPage: !!builderProfiles.find(bp => bp.toLowerCase() === address.toLowerCase()),
+              // Store the profile path for navigation
+              profilePath: `/builders/${address}`,
+              // Used real name if available, otherwise used Builder
+              name: builderNamesMap[address] || "Builder",
+              // Use profile image if available, otherwise fallback to generated avatar
+              avatarUrl: profileImageMap[address] || `https://effigy.im/a/${address}.svg`,
+            };
+        }),
+      );
+      setBuilders(
+        builders
+          .filter(builder => !!builder)
+          // Sort by personal page availability
+          .sort((a, b) => +(b.hasPersonalPage === true) - +(a.hasPersonalPage === true)),
+      );
+    };
+
+    fetchBuilders().catch(console.error);
+  }, [events, builderProfiles, builderNamesMap, profileImageMap]);
+
+  return (
+    <div className="min-h-screen bg-gradient-to-b from-base-200 to-base-300">
+      <div className="container mx-auto px-4 py-12" ref={elementRef}>
+        {/* Hero Section */}
+        <div className="text-center mb-16">
+          <h1 className="text-5xl font-extrabold mb-4 bg-gradient-to-r from-purple-600 via-pink-500 to-blue-500 text-transparent bg-clip-text">
+            Batch 14 Builders
+          </h1>
+        </div>
+
+        {/* Builders Grid */}
+        {isLoading ? (
+          <div className="flex flex-col items-center justify-center py-12">
+            <div className="animate-spin rounded-full h-16 w-16 border-b-2 border-primary mb-4"></div>
+            <p className="text-xl font-medium">Loading builders...</p>
+          </div>
+        ) : (
+          <div className="grid grid-cols-1 md:grid-cols-3 lg:grid-cols-4 gap-4">
+            {builders.map((builder, id) => (
+              <div
+                key={id}
+                className="card bg-base-100 shadow hover:shadow-md transition-all duration-300 overflow-hidden border border-transparent hover:border-primary"
+              >
```

Comment:
You should only have hover effects for interactive elements. This hover effect misleads the user into thinking they can click it to navigate to the profile, while in reality, only the "View" button is clickable.